File: postgraduate/src/main/webapp/js/controller/selCourseController.js
app.controller('selCourseController',function($scope,$http,selCourseService){
	//查询全部选课情况
	$scope.selectAll=function(id){
		selCourseService.selectAll(id).success(
				function(response){
					$scope.list=response;
					console.log(response);
				}	
		);
	}
	//查询全部选课情况
	$scope.adminSelAll=function(){
		selCourseService.adminSelAll().success(
				function(response){
					$scope.adminlist=response;
				}	
		);
	}
	//增加保存
	
	$scope.save=function(uid,sid){
		var object=null;		
			object=selCourseService.insert(uid,sid);
		
		object.success(
				function(response){
					if(response.success){
						
					}else{
						alert(response.message);
					}
				}
		);
	}
	//查询实体
	$scope.selectOneLes=function(id){
		selCourseService.selectOne(id).success(
			function(response){
				$scope.lentity=response;
			}		
		);
	} 
	$scope.studelete=function(stuid,lesteaid){
		selCourseService.studelete(stuid,lesteaid);
	}
	$scope.state=['未审核','已选中','未选中'];
});

File: postgraduate/src/main/java/com/postgraduate/controller/SelCourseController.java
package com.postgraduate.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;


import com.postgraduate.pojo.TbSelCourse;
import com.postgraduate.service.SelCourseService;

import entity.Result;

@RestController
@RequestMapping("/SelCourse")
public class SelCourseController {
	@Autowired
	private SelCourseService selCourseService;
	/**
	 * 返回全部列表
	 * 
	 */
	@RequestMapping("/selectAll")
	public List<TbSelCourse> selectAll(int id){
		return selCourseService.selectAll(id);
	}
	/**
	 * 管理员查询全部选课情况
	 * 
	 */
	@RequestMapping("/adminSelAll")
	public List<TbSelCourse> adminSelAll(){
		return selCourseService.adminSelAll();
	}
	/**
	 * 增加学生
	 * @return
	 */
	@RequestMapping("/insert")
	public Result insert(int uid ,int sid) {
		try {
			selCourseService.insert(uid,sid);
			return new Result(true,"增加成功");
		}catch(Exception e) {
			e.printStackTrace();
			return new Result(false,"增加失败");
		}
	}
	/**
	 * 删除已选课程
	 * @return
	 */
	@RequestMapping("/delete")
	public Result delete(int stuid ,int lesteaid) {
		try {
			selCourseService.delete(stuid,lesteaid);
			return new Result(true,"增加成功");
		}catch(Exception e) {
			e.printStackTrace();
			return new Result(false,"增加失败");
		}
	}
}
